pkgs/development/tools/build-managers/cmake/default.nix

```diff
@@ -20,18 +20,21 @@
 , SystemConfiguration
 , ps
 , isBootstrap ? false
-, useNcurses ? false
 , useOpenSSL ? !isBootstrap
 , useSharedLibraries ? (!isBootstrap && !stdenv.isCygwin)
-, withQt5 ? false
-, buildDocs ? (!isBootstrap && (useNcurses || withQt5))
+, uiToolkits ? [] # can contain "ncurses" and/or "qt5"
+, buildDocs ? !(isBootstrap || (uiToolkits == []))
 }:
 
+let
+  cursesUI = lib.elem "ncurses" uiToolkits;
```

Comment:
I'm not sure I agree that `uiToolkits` list instead of booleans is a cleaner interface, but no strong feelings. But, more importantly, it seems the current implementation doesn't warn the user of typos, which I think is bad UX. Please check uidToolkits for valid elements?

Reply:
My "strong feeling", to be honest, is the lack of consistency: `useNcurses` and `withQt` are completely differentm and it betrays completely the idea they should convey.

Ok, I will implement this as an assertion.